cypress/integration/dib-travel/other/generic-product.spec.ts

```diff
@@ -0,0 +1,341 @@
+import { GenericProduct } from '@cy/models';
+import { addGenericProduct, confirmAddedGenericProduct, deleteGenericProduct, tooltip } from './helpers';
+
+describe('Other - Generic Product', () => {
+  let genericProduct: GenericProduct;
+
+  const selectForAddingGenericProduct = (containerItem: string): void => {
+    cy.get('home dib-generic-product-form ui-control-wrapper label')
+      .contains(containerItem)
+      .parents('ui-control-wrapper')
+      .click();
+  };
+
+  const selectForEditingGenericProduct = (containerItem: string): void => {
+    cy.get('.cdk-overlay-container dib-generic-product-form ui-control-wrapper label')
+      .contains(containerItem)
+      .parents('ui-control-wrapper')
+      .click();
+  };
+
+  before(() => {
+    cy.fixture('other/generic-product').then((genericProductFixture) => {
+      genericProduct = genericProductFixture;
+    });
+  });
+
+  beforeEach(() => {
+    cy.loginAgent();
+
+    cy.get('home dib-product-pickers dib-tabs-header .label').contains('Other').click();
+  });
+
+  it('should display "Other" in the navbar', () => {
+    cy.get('home dib-product-pickers dib-tabs-header .label').contains('Other').should('exist');
+  });
+
+  it('should display Other page', () => {
+    cy.get('home dib-generic-product-form button').contains('Add to cart').should('exist');
+  });
+
+  it('should display link for restrictions before booking', () => {
+    cy.get('home dib-covid-banner span').should(
+      'contain',
+      'Get the latest travel and health restrictions before booking your trip.'
+    );
+    cy.get('home dib-covid-banner a[href="https://apply.joinsherpa.com/travel-restrictions"]').should(
+      'have.text',
+      ' Find out now '
+    );
+  });
+
+  it('should not be able to submit an empty generic product form', () => {
+    cy.get('home dib-generic-product-form ui-button[type=primary]').click();
+
+    cy.get('home dib-generic-product-form ui-input .error').should('contain', 'This field is required');
+  });
+
+  it('should check content of tooltips for generic product form', () => {
+    tooltip(0);
+
+    cy.get('.tooltip-content').should(
+      'contain',
```

Comment:
'have.text'

Reply:
@sjova I talked to Stefan and in some places like here we can't use `have.text` instead of `contain`, because some sections of the text aren't well separated.